src/schema/posts.ts

```diff
@@ -1550,6 +1667,96 @@ export const resolvers: IResolvers<any, Context> = {
       });
       return { _: true };
     },
+    votePost: async (
+      source,
+      { id, vote }: { id: string; vote: UserPostVote },
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      try {
+        if (!Object.values(UserPostVote).includes(vote)) {
+          throw new ValidationError('Unsupported vote type');
+        }
+
+        const post = await ctx.con.getRepository(Post).findOneByOrFail({ id });
+        await ensureSourcePermissions(ctx, post.sourceId);
+        const userPostRepo = ctx.con.getRepository(UserPost);
+
+        switch (vote) {
+          case UserPostVote.Up:
+            await userPostRepo.save({
+              postId: id,
+              userId: ctx.userId,
+              vote: UserPostVote.Up,
+              hidden: false,
+            });
+
+            break;
+          case UserPostVote.Down:
+            await userPostRepo.save({
+              postId: id,
+              userId: ctx.userId,
+              vote: UserPostVote.Down,
+              hidden: true,
+            });
+
+            break;
+          case UserPostVote.None:
+            await userPostRepo.save({
+              postId: id,
+              userId: ctx.userId,
+              vote: UserPostVote.None,
+              hidden: false,
+            });
+
+            break;
+          default:
+            throw new ValidationError('Unsupported vote type');
+        }
+      } catch (err) {
+        // Foreign key violation
+        if (err?.code === TypeOrmError.FOREIGN_KEY) {
+          throw new NotFoundError('Post or user not found');
+        }
+
+        throw err;
+      }
+
+      return { _: true };
+    },
+    dismissPostFeedback: async (
```

Comment:
Happy with this for now, down the line we probably would like a generic function inside as we do for post flags.